sdk/eventhub/azure-eventhub/azure/eventhub/_consumer.py

```diff
@@ -188,19 +199,26 @@ def _open(self):
         if not self.running:
             if self._handler:
                 self._handler.close()
-            auth = self._client._create_auth()
-            self._create_handler(auth)
-            self._handler.open(
-                connection=self._client._conn_manager.get_connection(
-                    self._client._address.hostname, auth
-                )  # pylint: disable=protected-access
+            # TODO: using JWTTokenAuth not working 
+            #auth = self._client._create_auth()
+            eventhub_name = self._client._address.path.strip("/")
+            hostname = self._client._address.hostname
+            uri = "sb://{}/{}".format(hostname, eventhub_name)
+            key = self._client._credential.key
+            policy = self._client._credential.policy
+            auth = SASTokenAuth(
+                uri=uri,
+                audience=uri,
+                username=policy,
+                password=key
             )
```

Comment:
we could reuse `self._create_auth` to create the authentication object that is required.

Reply:
for some reason JWTTokenAuth wasn't working, it was on my TODO list. however, I'll try your snippet above